src/internal/kopia/upload.go

```diff
@@ -650,10 +649,33 @@ func traverseBaseDir(
 	return nil
 }
 
+// TODO(ashmrtn): We may want to move this to BackupOp and pass in
```

Comment:
Alternatively, I think this should be in the path package, sans the Reason struct.  Something like his:

```go
func NewBuilder(tenant, owner string, s ServiceType, c CategoryType, append string...) (*path.Builder, error) {...}
```

We do a fair bit of `path.Builder{}.Append("...").ToSomePath()` throughout Corso.  This feels like a natural QoL.  Run the "tmp">.Dir() version as shown if "append" is nil, and we've got both cases covered.

Otherwise, I agree about having operations pass in the premade paths.

Reply:
I'll make a follow up PR to move making the path and passing it to BackupCollections as it adds more code and there's already a lot going on in this PR. I did make #1832 to track it though. I will probably hold off a bit on revisiting the path package for QoL stuff right now